scipy/stats/_continuous_distns.py

```diff
@@ -5914,28 +5914,29 @@ def _isf(self, q, df):
         return -sc.stdtrit(df, q)
 
     def _stats(self, df):
-        # 0.0, 1.0, 0.0, 0.0
-        cond = df == np.inf
-        mu = np.where(df > 1 | cond, 0.0, np.inf)
-
-        def _mu2_finitedf(df):
-            mu2 = _lazywhere(df > 2, (df,),
-                             lambda df: df / (df-2.0),
-                             np.inf)
-            mu2 = np.where(df <= 1, np.nan, mu2)
-            return mu2
-
-        mu2 = _lazywhere(~cond, (df,), _mu2_finitedf, 1.0)
-        g1 = np.where(df > 3 | cond, 0.0, np.nan)
-
-        def _g2_finitedf(df):
-            g2 = _lazywhere(df > 4, (df,),
-                            lambda df: 6.0 / (df-4.0),
-                            np.inf)
-            g2 = np.where(df <= 2, np.nan, g2)
-            return g2
+        # infinite df -> normal distribution (0.0, 1.0, 0.0, 0.0)
+        infinite_df = np.isposinf(df)
+
+        mu = np.where(df > 1, 0.0, np.inf)
+
+        condlist = ((df > 1) & (df <= 2),
+                    (df > 2) & np.isfinite(df),
+                    infinite_df)
+        choicelist = (lambda df: np.broadcast_to(np.inf, df.shape),
+                      lambda df: df / (df-2.0),
+                      lambda df: np.broadcast_to(1, df.shape))
+        mu2 = _lazyselect(condlist, choicelist, (df,), np.nan)
```

Comment:
Oh, I wasn't aware of the `_lazyselect` function. This looks much cleaner, thanks!